calcom/cal.com
# fix: browser back button not working

## What does this PR do?
Fixes a bug due to which the back button on the booking page was not working.
The issue was due to multiple calls to `setQueryParams` and `removeQueryParams` as it was pushing multiple URLs to the browser history.  

## My fix
Instead of pushing to the history I am now replacing the last entry in the history.

Fixes #13324 


https://github.com/calcom/cal.com/assets/55000107/79f9645e-2264-4f47-bb85-d00c81fec69d

## Type of change
- Bug fix (non-breaking change which fixes an issue)


## How should this be tested?
- Go to the booking page of a user with multiple events present. 
- Open one of the events and change the date selected.
- Press the browser back button (You should be taken back to the user page.)
- Try repeating this with the other event .


## Mandatory Tasks

- [ x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.
